pkg/core/resources/model/resource.go

```diff
@@ -777,7 +779,9 @@ type TypedResourceIdentifier struct {
 
 type ResolvedBackendRef struct {
 	LegacyBackendRef *common_api.BackendRef
```

Comment:
If these are mutually exclusively set, is it possible to have a private interface like 

```
type IsResolvedBackendRef interface {
  isResolvedBackendRef()
}
```

and then have

```
type ResolvedBackendRef struct {
  Weight uint
  Ref IsResolvedBackendRef
}

...

var ref ResolvedBackendRef

switch t := ref.Ref {
case *LegacyBackendRef:
case *RealResourceBackendRef:
}
```

so that we always have exactly one of these?

Reply:
or otherwise make it clear via comments what it means when one thing is set/not set and whether both can be set

Reply:
I did like you suggested with a single `Ref` field for these mutually exclusive values. It's actually nice that now we have several functions like `SniForBackendRef` and `ServiceTagIdentities` that accept only `RealResourceBackendRef` and don't require checking for legacy stuff.